### PR TITLE
Don't keep a strong ref in storage for destroyed resources

### PR DESCRIPTION
**Pull request by gents83**
Changing Destroyed(Arc<T>, Epoch) into Destroyed(Epoch) to actually release that reference

### CODE REVIEW

**Comment by gents83** [comment]
Depending on #4782

File: wgpu-core/src/storage.rs [hunk 1]
```diff
@@ -213,15 +213,12 @@ where
                 std::mem::replace(slot, Element::Vacant)
```
Comment (nical) [inline wgpu-core/src/storage.rs]: Nit: with this change you can make this even simpler, no need for the borrow check workaround, you can `std::mem::replace(slot, Element::Destroyed(storage_epoch))` directly without the intermediate `Vacant` step.

**Review by nical — APPROVED** [review]
Let's merge #4782 first, then rebase this on top of land it.

**Review by nical — APPROVED** [review]
Neat!

**Pull request merged**